06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndexClass.js

```diff
@@ -0,0 +1,133 @@
+/**
+ *
+ *
+ * @class InvertedIndex
+ */
+class InvertedIndex {
+/**
+* Creates an instance of InvertedIndex.
+* @memberOf InvertedIndex
+*/
+  constructor() {
+    this.mainIndex = {};
+    this.bookNames = [];
+  }
+
+/**
+ * Checks input to see if it comforms to specific standards.
+ * @param {Array} book - Book file to be validated
+ * @returns {boolean} true/false - returns validation status.
+ * @memberOf InvertedIndex
+ */
+  validateInput(book) {
+    if (Array.isArray(book) && book.length > 0 && typeof book[0] === 'object') {
+      if (book[0].text && book[0].title) {
+        if (typeof (book[0].text) === 'number') {
+          return false;
+        }
+        return true;
+      }
+    }
+    return false;
+  }
+
+
+/**
+ * Gets a book and returns it as as a concatenated text
+ * @param {Object} book - a book object with title and text property
+ * @returns {String} text- returns boolean if conditions are not met
+ * @memberOf InvertedIndex
+ */
+  getBookAsText(book) {
+    const status = this.validateInput([book]);
+    if (status) {
+      this.bookNames.push(book.title);
+      return `${book.title} ${book.text}`;
+    }
+    return status;
+  }
+
+/**
+* Removes characters, whitespaces and converts text to array elements.
+* @param {String} text returned from getBookAsText
+* @returns {Array} -returns an array of words in lower-cases with no characters
+* @memberOf InvertedIndex
+*/
+  generateToken(text) {
+    return text.toLowerCase()
+    .replace(/[^\w\s]|_/g, '')
+    .split(/\s+/);
+  }
+
+
+/**
+ * Gets an array of Words and makes element have unique occurences
+ * @param {Object} arrayOfWords - an  book object with title and text property
+ * @returns {Array} arrayOfWords - a filtered array with unique elements
+ * @memberOf InvertedIndex
+ */
+  returnUniqueWords(arrayOfWords) {
```

Comment:
How about avoiding including javascript data types in your variable naming - 'arrayOfWords' could just be 'words' ?